src/prayer/PrayerList/index.js

```diff
@@ -61,152 +61,134 @@ const Footer = styled(({ theme }) => ({
   state = {
-    prayers: [],
     prayerIndex: 0,
     prayed: false,
-    saved: false,
   };
 
-  title = this.props.navigation.getParam('title', 'My Church');
-
-  componentDidMount() {
-    this.setState(() => {
-      const prayers = this.props.navigation.getParam(
-        'prayers',
-        cache.readQuery({ query: GET_PRAYERS }).prayers
-      );
-
-      return {
-        prayers,
-        saved: prayers[0].isSaved,
-      };
-    });
-  }
-
   static navigationOptions = {
     header: null,
   };
 
   render() {
-    const prayer = this.state.prayers[this.state.prayerIndex];
-    const isLastPrayer =
-      this.state.prayerIndex + 1 === this.state.prayers.length;
-
-    const advancePrayer = (prayed = false) =>
-      !isLastPrayer
-        ? this.setState((prevState) => ({
-            prayerIndex: prevState.prayerIndex + 1,
-            prayed: prayed ? false : prevState.prayed,
-            saved: prevState.prayers[prevState.prayerIndex + 1].isSaved,
-          }))
-        : this.props.navigation.popToTop();
+    const title = this.props.navigation.getParam('title', 'My Church');
+    const type = this.props.navigation.getParam('type', null);
 
     return (
       <ModalView onClose={() => this.props.navigation.popToTop()}>
         <FlexedSafeAreaView>
-          <Mutation
-            mutation={FLAG_PRAYER}
-            update={() => flagPrayerUpdateAll(cache, prayer.id)}
+          <Query
+            query={GET_PRAYER_FEED}
+            variables={{ type, first: 20 }}
+            fetchPolicy={'cache-and-network'}
           >
-            {(flagPrayer) => (
-              <Mutation mutation={INCREMENT_PRAYER_COUNT}>
-                {(increment) => (
-                  <FlexedView>
-                    <ScrollArea>
-                      <ScrollView>
-                        <Header>
-                          <H6>Praying for</H6>
-                          <GreenH4>{this.title}</GreenH4>
-                        </Header>
-                        <StyledPrayerView>
-                          {prayer ? (
-                            <PrayerSingle
-                              avatarSize={'medium'}
-                              navigation={this.props.navigation}
-                              prayer={prayer}
-                              action={
-                                <SaveButton
-                                  toggleSavedState={() =>
-                                    this.setState((prevState) => ({
-                                      saved: !prevState.saved,
-                                    }))
-                                  }
-                                  saved={this.state.saved}
-                                  prayerID={prayer.id}
+            {({ data, loading, error }) => {
+              if (loading) return <ActivityIndicator />;
+              if (error) return <ErrorCard />;
+              const prayers = data.prayerFeed.edges.map((edge) => edge.node);
```

Comment:
Is it possible for prayerFeed to be undefined?

Reply:
don't think so. but I'm wrong a lot...